### PR TITLE
Test on Alpine, and make tests pass

### LINKED ISSUE #1499: test_getaddrinfo failure
I'm running the tests on Alpine Linux with Python 3.8 and all the newest dependencies as of the time of writing. All tests succeed except for `test_getaddrinfo` in `trio/tests/test_socket.py`:

```
============================================================= FAILURES ==============================================================
_________________________________________________________ test_getaddrinfo __________________________________________________________

monkeygai = <trio.tests.test_socket.MonkeypatchedGAI object at 0x7fcd6388ff10>

    async def test_getaddrinfo(monkeygai):
        def check(got, expected):
            # win32 returns 0 for the proto field
            def without_proto(gai_tup):
                return gai_tup[:2] + (0,) + gai_tup[3:]
    
            expected2 = [without_proto(gt) for gt in expected]
            assert got == expected or got == expected2
    
        # Simple non-blocking non-error cases, ipv4 and ipv6:
        with assert_checkpoints():
            res = await tsocket.getaddrinfo(
                "127.0.0.1", "12345", type=tsocket.SOCK_STREAM
            )
    
>       check(res, [
            (tsocket.AF_INET,  # 127.0.0.1 is ipv4
             tsocket.SOCK_STREAM,
             tsocket.IPPROTO_TCP,
             "",
             ("127.0.0.1", 12345)),
        ])  # yapf: disable

trio/tests/test_socket.py:116: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

got = [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '127.0.0.1', ('127.0.0.1', 12345))]
expected = [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 12345))]

    def check(got, expected):
        # win32 returns 0 for the proto field
        def without_proto(gai_tup):
            return gai_tup[:2] + (0,) + gai_tup[3:]
    
        expected2 = [without_proto(gt) for gt in expected]
>       assert got == expected or got == expected2
E       AssertionError: assert ([(<AddressFam...0.1', 12345))] == [(<AddressFam...0.1', 12345))]
E         At index 0 diff: (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '127.0.0.1', ('127.0.0.1', 12345)) != (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 12345))
E         Use -v to get the full diff or [(<AddressFam...0.1', 12345))] == [(<AddressFam...0.1', 12345))]
E         At index 0 diff: (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '127.0.0.1', ('127.0.0.1', 12345)) != (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 0, '', ('127.0.0.1', 12345))
E         Use -v to get the full diff)

trio/tests/test_socket.py:108: AssertionError
```

### PR DESCRIPTION
**Pull request by njsmith**
Fixes #1499

### CODE REVIEW

File: trio/tests/test_socket.py [hunk 1]
```diff
@@ -101,11 +101,19 @@ def test_socket_has_some_reexports():
 async def test_getaddrinfo(monkeygai):
     def check(got, expected):
         # win32 returns 0 for the proto field
-        def without_proto(gai_tup):
-            return gai_tup[:2] + (0,) + gai_tup[3:]
+        # alpine and glibc have inconsistent handling of the canonical name
```
Comment (PureTryOut) [inline trio/tests/test_socket.py:104]: This should probably be the following:

```suggestion
        # musl and glibc have inconsistent handling of the canonical name
```

If libc is the issue, then Musl is causing it, not Alpine :wink:

**Comment by njsmith** [comment]
Guess this is working and ready for review.

Merging will be a bit weird because it changes the set of CI jobs – I guess once someone approves it, I'll go change the set of required checks and then hit the green button myself?

**Comment by PureTryOut** [comment]
Awesome, thanks for this! Fixing it one-time is one thing, but actually adding it to your CI and making sure it'll work in the future is another! It makes my live as a packager so much better, I wish every project did this. Thank you _very_ much!

File: trio/tests/test_socket.py (hunk 1 above)
Comment (njsmith) [reply trio/tests/test_socket.py]: Fair enough :-)

**Comment by PureTryOut** [comment]
I'd love too, but I don't currently see where you guys mention platform support? https://trio.readthedocs.io/en/stable/index.html mentions:

> Supported environments: Linux, macOS, or Windows running some kind of Python 3.5-or-better (either CPython or PyPy3 is fine). *BSD and illumos likely work too, but are untested.

Linux would seem good enough there though.

**Comment by pquentin** [comment]
Well, the tests do pass, but you need to update the list of required checks

**Comment by pquentin** [comment]
@PureTryOut if it's good enough for you then it's good enough. Thanks for opening that issue!

**Pull request merged**